custom_components/waste_collection_schedule/waste_collection_schedule/source/landkreis_rhoen_grabfeld.py

```diff
@@ -0,0 +1,51 @@
+import datetime
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "Source for Rhön Grabfeld"
+DESCRIPTION = "Source for Rhönn Grabfeld uses service by offizium."
+URL = 'https://fs-api-rg.offizium.com/abfalltermine'
+TEST_CASES = {
+    "City only": {"city": "Ostheim"},
+    "City + District": {"city": "Ostheim", "district": "Oberwaldbehrungen"},
+    "District only": {"district": "Oberwaldbehrungen"},
+    "empty": {}
+}
+
+
+class Source:
+    def __init__(self, city: str = None, district: str = None):
+        self._city = city
+        self._district = district
+        self._iconMap = {
+            "Restmüll/Gelber Sack/Biotonne": "mdi:trash-can",
+            "Papiersammlung": "mdi:package-variant",
+            "Problemmüllsammlung": "mdi:biohazard"
+        }
+
+    def fetch(self):
+        now = datetime.datetime.now().date()
+
+        r = requests.get(URL, params={
+            "stadt": self._city,
+            "ortsteil": self._district
+        })
+
+        r.raise_for_status()
+
+        entries = []
+        for event in r.json():
+            # filter out Sammelstellen, Wertstoffhof and Wertstoffzentrum
+            if event["muellart"] in self._iconMap:
```

Comment:
Not sure if it is a good idea to use the icon-map to filter out waste types. I think I would prefer a blacklist.